Makefile

```diff
@@ -43,20 +48,40 @@ build:
 		$(SED) -i "/^go 1/a replace github.com/dnsimple/coredns-dnsimple => $$PLUGIN_DIR" go.mod; \
 		$(SED) -i '/route53:route53/i dnsimple:github.com\/dnsimple\/coredns-dnsimple' plugin.cfg; \
 	fi; \
-	GOFLAGS=-mod=mod go generate; \
-	go mod tidy; \
 	gitcommit=$(shell git describe --dirty --always); \
 	go build -o coredns -ldflags="-s -w -X github.com/coredns/coredns/coremain.GitCommit=$$gitcommit" .
-	@cp coredns/coredns coredns-dnsimple
+	@mkdir -p bin
+	@cp coredns/coredns bin/coredns-dnsimple
+
+.PHONY: docker-build
+docker-build:
+	@echo "Building Docker image"
+	@if [ ! -d "coredns.docker" ]; then \
+		git clone https://github.com/coredns/coredns.git coredns.docker; \
+	fi
+	cd coredns.docker; \
+	if ! grep -q "replace github.com/dnsimple/coredns-dnsimple" go.mod; then \
+		$(SED) -i "/^go /a replace github.com/dnsimple/coredns-dnsimple => ./plugin/dnsimple" go.mod; \
+		$(SED) -i '/route53:route53/i dnsimple:github.com\/dnsimple\/coredns-dnsimple' plugin.cfg; \
+	fi; \
+	mkdir -p plugin/dnsimple; \
+	cp ../*.go plugin/dnsimple; \
+	cp ../go.mod plugin/dnsimple; \
+	GOFLAGS=-mod=mod go generate; \
+	go mod tidy; \
+	cp ../Dockerfile.release Dockerfile; \
+	cp ../bin/_docker/docker-entrypoint.sh docker-entrypoint.sh; \
+	rm .dockerignore; \
+	docker build --no-cache --build-arg PACKAGER_VERSION=$(PKG_VERSION) --no-cache -t dnsimple/coredns:$(PKG_VERSION) .
```

Comment:
It looks that we should also copy the `.tool-versions` file, right?  

```
	docker build --no-cache --build-arg PACKAGER_VERSION=1.3.0 --no-cache -t dnsimple/coredns:1.3.0 .
No version is set for command go
Consider adding one of the following versions in your config file at ~/coredns-dnsimple/coredns.docker/.tool-versions
golang 1.24.0
golang 1.23.6
golang 1.23.5
golang 1.23.4
golang 1.23.2
No version is set for command go
Consider adding one of the following versions in your config file at /Users/san983/dnsimple/coredns-dnsimple/coredns.docker/.tool-versions
golang 1.24.0
golang 1.23.6
golang 1.23.5
golang 1.23.4
golang 1.23.2
```

Reply:
Is this related to asdf not exporting the correct version locally? I tested with a devcontainer on go 1.23.4 and was able to run both `build` & `build-docker` without receiving the error